pkg/ddc/thin/engine.go

```diff
@@ -124,15 +132,39 @@ func CheckReferenceDatasetRuntime(client client.Client, runtime *datav1alpha1.Th
 	var mounted []types.NamespacedName
 	if dataset != nil {
 		// getMountedDataset from dataset first
+		ctx.Log.V(1).Info("Get physical dataset from virtual dataset mounts")
 		mounted = base.GetPhysicalDatasetFromMounts(dataset.Spec.Mounts)
 	} else if runtime.Status.Mounts != nil && len(runtime.Status.Mounts) != 0 {
 		// then try to getMountedDataset from runtime
+		ctx.Log.V(1).Info("Virtual dataset not found, try to get physical dataset from runtime mounts")
 		mounted = base.GetPhysicalDatasetFromMounts(runtime.Status.Mounts)
 	}
 	// not mount other datasets
 	if len(mounted) == 0 {
 		return false, nil
 	}
 
+	// patch runtime with reference annotation
+	_, err = PatchReferenceThinRuntimeAnnotation(ctx.Client, runtime)
```

Comment:
I suggest moving PatchReferenceThinRuntimeAnnotation to another PR.

Reply:
OK